openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -274,6 +274,27 @@ def get_from_id(cls, course_id):
 
         return course_overview or cls.load_from_module_store(course_id)
 
+    @classmethod
+    def get_from_ids_if_exists(cls, course_ids):
+        """
+        Return a dict mapping course_ids to CourseOverviews, if they exist.
+
+        This method will *not* generate new CourseOverviews or delete outdated
+        ones. It exists only as a small optimization used when CourseOverviews
+        are known to exist, for common situations like the student dashboard.
+
+        Callers should assume that this list is incomplete and fall back to
+        get_from_id if they need to guarantee CourseOverview generation.
+        """
+        return {
+            overview.id: overview
+            for overview
+            in cls.objects.select_related('image_set').filter(
```

Comment:
This will load the `image_set` if it exists. It's possible to get a `CourseOverview` with no images generated for it, in which case it'll lazily make the appropriate thumbnails -- though that happens at publish time, so it's unlikely. Image sets are 1:1 with `CourseOverview` objects, they're just an optional piece that was added later.